src/main/scala/wdlTools/types/Stdlib.scala

```diff
@@ -159,57 +156,55 @@ case class Stdlib(conf: Options, version: WdlVersion) {
   // evaluate the output type of a function. This may require calculation because
   // some functions are polymorphic in their inputs.
   private def evalOnePrototype(funcDesc: T_Function,
-                               inputTypes: Vector[T],
-                               text: TextSource,
-                               docSourceUrl: Option[URL]): Option[(T, T_Function)] = {
+                               inputTypes: Vector[T]): Either[Option[(T, T_Function)], String] = {
```

Comment:
This code is already complex, and it isn't done yet. `Either` is not going to help it become easier to understand, and it will surely become more involved later on.